### ISSUE TITLE
Authentication against MongoDB via SCRAM-SHA-1

### ISSUE THREAD

**Issue by sourcekick** [scullxbones/akka-persistence-mongo#27]
Please support the new database user authentication method supported by MongoDB called SCRAM-SHA-1.

The MongoDB drivers have now been updated to support this authentication method:
http://docs.mongodb.org/manual/release-notes/3.0-scram/#considerations-scram-sha-1-drivers
The relevant lowest casbah Scala Driver version is 2.8.0 according to the listing.

Here is the relevant API documentation of casbah that explains the how-to:
http://mongodb.github.io/casbah/guide/connecting.html#authentication

You just need to add some configuration property and use the following factory method for SCRAM-SHA-1:
`createScramSha1Credential`

As far as I can see, the small change would have to be made in:
https://github.com/scullxbones/akka-persistence-mongo/blob/ae092d0929fd13e1ab45a043d539b9a572ce782a/casbah/src/main/scala/akka/contrib/persistence/mongodb/CasbahPersistenceExtension.scala

**Comment by sourcekick** [comment]
Current code:
```
  private[mongodb] lazy val client =
    userPass.map {
      case (u,p) => authenticated(u,p)
    }.getOrElse(unauthenticated())

  private[mongodb] lazy val db = client(mongoDbName)

  private[this] def authenticated(user: String, password: String)
= MongoClient(mongoUrl,MongoCredential.createMongoCRCredential(user,mongoDbName,password.toCharArray) :: Nil)
```

**Comment by scullxbones** [comment]
I'll save this for when I address mongo3 compatibility.  This will wait on reactivemongo 0.11 to be released.

**Comment by ulrichwinter** [comment]
+1

**Comment by WalkingGhost1975** [comment]
+1

**Comment by mojadev** [comment]
+1

**Comment by richard-mueller** [comment]
+1

**Comment by ulrichwinter** [comment]
I would suggest to add an URI-style db configuration, too. 
This is the most flexible configuration mechanism, because it allows to specify all available options (including the authentication mechanism):
http://docs.mongodb.org/manual/reference/connection-string/

The casbah-driver supports it too:
http://mongodb.github.io/casbah/guide/connecting.html#id2

**Comment by scullxbones** [comment]
I actually *think* this is already supported without a code change.  Can you try using a mongodb uri something like the following:
```
akka.contrib.persistence.mongodb.mongo.urls = [ "mongodb://mongouser:mongopassword@mongohost/?authMechanism=SCRAM-SHA-1" ] 
```

**Comment by scullxbones** [comment]
As i'm going back through this, testing against mongo 3, those urls are definitely not supported.

**Comment by scullxbones** [comment]
Fixed with 0.4.0
